### DISCUSSION TITLE
[Feature] Debugger

### DISCUSSION THREAD

**Discussion by AMatutat** [Programmiermethoden/Dungeon#1373] [Ideas]
Im Dungeon ist es schwer richtig zu Debuggen. 

Spielinhalte laufen in der GameLoop und diese läuft (per Default) 30-mal pro Sekunde. Hier mit klassischen Debuggen durchzugehen, ist sehr nervig. Außerdem müssen vorher erst bestimmte Situationen ausgelöst werden (teilweise sind diese ja auch zufällig). 

Ideen, die mal in den Raum geworfen wurde:
- Verhaltensbasiertes Debuggen
- Eigenes Debugger-Tool als IntelliJ Plugin (verschmelzen mit dem IntelliJ Debugger) 
- Debug Menü
- Live Level Editor zum erzeugen bestimmter Spielsituationen.

**Comment by cagix** [comment]
mögliche interessante Quelle: https://microsoft.github.io/debug-adapter-protocol/
